tchannel/tornado/peer.py

```diff
@@ -409,6 +411,17 @@ def _send(self, connection, req):
         with timeout(response_future, req.ttl):
             try:
                 response = yield response_future
+            except StreamClosedError as error:
+                error = NetworkError(
+                    id=req.id,
+                    description=error.message,
+                    tracing=req.tracing,
+                )
+                # event: after_receive_error
+                self.tchannel.event_emitter.fire(
+                    EventType.after_receive_error, req, error,
+                )
+                raise
```

Comment:
I'm a little confused here. Are you trying to re-raise the `StreamClosedError` you caught or do you want the new `NetworkError` to be raised?